src/MPAGSCipher/PlayfairCipher.cpp

```diff
@@ -0,0 +1,171 @@
+    }
+
+    for (auto p : playfairciphermapinv_){
+        std::cout << p.first.first << ":" << p.first.second << " :" << p.second << std::endl;
+    }
+    */
+    
+}
+
+
+std::string PlayfairCipher::applyCipher(const std::string& inputText,
+                                        const CipherMode cipherMode) const {
+
+    std::string output{""},modifiedinput{""},input{inputText};
+
+    switch(cipherMode){
+        case CipherMode::Encrypt:
+            // Change J → I
+            std::transform(input.begin(),input.end(),input.begin(),[](char& val){if(val=='J'){return 'I';} else {return val;}});
+            std::cout << "after j->i: " << input << std::endl;
+            // If repeated chars in a digraph add an X or Q if XX
+            modifiedinput += input[0];
+            for(std::size_t j{1};j<input.size();j++){
+                if(input[j] == input[j-1]){
+                    if(input[j-1] != 'X'){
+                        modifiedinput += 'X';
+                        modifiedinput += input[j];
+                    } else {
+                        modifiedinput += 'Q';
+                        modifiedinput += input[j];
+                    }
+                } else {
+                    modifiedinput += input[j];
+                }
+            }
+            std::cout << "after double char correction: " << modifiedinput << std::endl;
+            // if the size of input is odd, add a trailing Z
+            if(modifiedinput.size()%2 != 0){
+                modifiedinput += 'Z';
+            }
+            std::cout << "after Z: " << modifiedinput << std::endl;
+            // Loop over the input in Digraphs
+            for(std::size_t k{0};k<modifiedinput.size();k+=2){
+                std::cout << "iteration var: " << k  << "chars: " << modifiedinput[k] << "+" << modifiedinput[k+1] << std::endl;
+                char digraphchar1{modifiedinput[k]};
+                char digraphchar2{modifiedinput[k+1]};
+
+                // - Find the coords in the grid for each digraph
+
+                auto iterindex1{playfairciphermap_.find(digraphchar1)};
+                auto iterindex2{playfairciphermap_.find(digraphchar2)};
+                std::pair<int,int> charcoord1{(*iterindex1).second};
+                std::pair<int,int> charcoord2{(*iterindex2).second};
+
+                std::cout << "coordinate char 1: " << charcoord1.first << ":" << charcoord1.second << std::endl;
+                std::cout << "coordinate char 2: " << charcoord2.first << ":" << charcoord2.second << std::endl;
+
+                // - Apply the rules to these coords to get 'new' coords
+
+                if(charcoord1.first == charcoord2.first){
+                    //same row rule
+                    std::cout << "row rule" << std::endl;
+                    if(charcoord1.second == 5){
+                        charcoord1.second = 1;
+                    } else {
+                        charcoord1.second++;
+                    }
```

Comment:
As I mentioned above, if these were zero indexed, you would just need to use the modulus:
```
charcoord1.second = (charcoord1.second + shift) % 5;
```